skiko/build.gradle.kts

```diff
@@ -427,6 +428,32 @@ kotlin {
     }
 }
 
+fun configureCinterop(
+    cinteropName: String,
+    os: OS,
+    arch: Arch,
+    target: KotlinNativeTarget,
+    targetString: String,
+    linkerOpts: List<String>,
+) {
+    val tasks = target.project.tasks
+    val taskNameSuffix = joinToTitleCamelCase(os.idWithSuffix(isIosSim = target.isIosSimArm64()), arch.id)
+    val writeCInteropDef = tasks.register("writeCInteropDef$taskNameSuffix", WriteCInteropDefFile::class.java) {
+        this.linkerOpts.set(linkerOpts)
+        outputFile.set(project.layout.buildDirectory.file("cinterop/$targetString/skiko.def"))
```

Comment:
Is this the reason why dynamic frameworks didn't work?

Reply:
The root issue is that the linker does not know it needs to look up macOS frameworks, when linking a dynamic framework. The linker opts can be configured by our Gradle plugin or via synthetic cinterop def file. We discussed available options with @pjBooms and decided that providing the cinterop file is a better solution.